test/name_core.jl

```diff
@@ -29,19 +30,26 @@ using Test
     end
 end
 
-@testset "combine_names" begin
-    @test combine_names((:a,), (:a,)) == (:a,)
-    @test combine_names((:a,:b), (:a,:b)) == (:a,:b)
-    @test combine_names((:a,:_), (:a,:b)) == (:a,:b)
-    @test combine_names((:a,:_), (:a,:_)) == (:a,:_)
+@testset "combine_names/combine_names_longest" begin
+    @test_throws DimensionMismatch combine_names((:a,), (:a, :b,))
 
-    @test combine_names((:a,:b,:c), (:_,:_,:_)) == (:a,:b,:c)
-    @test combine_names((:a,:_,:c), (:_,:b,:_)) == (:a,:b,:c)
-    @test combine_names((:_,:_,:_), (:_,:_,:_)) == (:_,:_,:_)
+    @test combine_names_longest((:a,), (:a, :b,)) == (:a, :b)
+    @test combine_names_longest((:a,), (:a,:_)) == (:a,:_)
+    @test combine_names_longest((:a,:b), (:a,:_,:c)) == (:a,:b,:c)
```

Comment:
Should probably include a DimensionMismatch test case for when the dimensions for `combine_names_longest((:a, :b), (:b, :a))` just for some self documenting test cases.